railties/test/generators/plugin_generator_test.rb

```diff
@@ -201,6 +228,12 @@ def test_action_mailer_is_removed_from_frameworks_if_skip_action_mailer_is_given
     assert_file "test/dummy/config/environments/production.rb" do |content|
       assert_no_match(/config\.action_mailer/, content)
     end
+    assert_no_directory "test/dummy/app/mailers"
+  end
+
+  def test_action_cable_is_removed_from_frameworks_if_skip_action_cable_is_given
+    run_generator [destination_root, "--skip-action-cable"]
+    assert_file "test/dummy/config/application.rb", /#\s+require\s+["']action_cable\/engine["']/
```

Comment:
```ruby
run_generator [destination_root, "--skip-action-cable"]
assert_file "test/dummy/config/application.rb", /#\s+require\s+["']action_cable\/engine["']/ passes
```` 
I think i got why it passes on master,

`test/dummy/` is generated by `Rails::Generators::AppGenerator` with filtered `options`,

https://github.com/rails/rails/blob/master/railties/lib/rails/generators/rails/plugin/plugin_generator.rb#L88-L104

then we replace some files in `test/dummy/` with all `options`
https://github.com/rails/rails/blob/master/railties/lib/rails/generators/rails/plugin/plugin_generator.rb#L106-L112

Next example doesn't pass on master, because these files are generated by `Rails::Generators::AppGenerator` with filtered `options`, https://github.com/rails/rails/blob/master/railties/lib/rails/generators/rails/plugin/plugin_generator.rb#L88-L104
```ruby
run_generator [destination_root, "--skip-action-cable"]
assert_no_file "test/dummy/config/cable.yml"
assert_no_file "test/dummy/app/assets/javascripts/cable.js"
assert_no_directory "test/dummy/app/channels"
assert_file "Gemfile" do |content|
  assert_no_match(/redis/, content)
end
```
it only will pass if add `:skip_action_cable` to `PASSTHROUGH_OPTIONS`. If don't pass `:skip_action_cable` to `PASSTHROUGH_OPTIONS` ,  these files will be always generated because `skip_action_cable` is `false` by default https://github.com/rails/rails/blob/master/railties/lib/rails/generators/app_base.rb#L55